### PR TITLE
Improve account `nonce` calculation for token-transfer transactions

### LINKED ISSUE #1686: Account `nonce` not updating properly on consecutive token-transfer transactions
### Expected behavior
The app should automatically update the `nonce` of an account after a transaction is performed, without the need to the user to refresh the whole application.

- A: Fetch latest nonce from onchain (useAuth)
- B: Fetch latest nonce from txpool (new hook)
- Math.max(A, B)

### Actual behavior
After 2 send tokens transactions, or switching the recipient account from one transaction to another, the transaction dry-run operation fail due to its nonce is lower than the account nonce.

https://user-images.githubusercontent.com/9727036/222161340-5c148bfd-9883-4e1a-bb51-e0f351352b9f.mp4

Example response getting from service on `POST /transactions/dryrun` call:

```js
{
  errorMessage: "Transaction with id:bfce3d5f39ab08c86675d2def37b7957eb4ef8fdb60fac8ecb3a2563101b164f nonce is lower than account nonce."
  events: []
  result: -1
}
```

### Steps to reproduce
1. Login to an account with balance.
2. Send tokens to another account.
3. Repeat the action 2 more times.
4. On the 3th try, the transaction dry-run will fail due to `nonce` not matching.

### Which version(s) does this affect? (Environment, OS, etc...)
All.

### PR DESCRIPTION
**Pull request by ClementeSerrano**
### What was the problem?

This PR resolves #1686

### How was it solved?

- [x] Add `useTransactionPoolQuery` hook.
- [x] Add `useAccountNonce` hook for last nonce calculation.
- [x] Fix `defaultTokenID` race condition on `Transaction` update of token:transfer. Now waiting `isTransactionSuccess` before updating default token ID.
- [x] Fix `signTransactionModal` infinite render loop due to `dryRunTransactionError` dep. Now reading dependency change from error message (scalar), not object.
- [x] Integrate new `useAuth` hook into `useCreateTransaction` hook in order to always fetch auth data with updated nonce.
- [x] Refetch account nonce on every transaction sign.
- [x] Increase UT coverage of token:transfer and auth hooks.

### How was it tested?

- [x] iOS Simulator.
- [x] Android Emulator.

### CODE REVIEW

File: src/modules/Accounts/api/useAccountPoolTransactionsQuery.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Fetch the transactions in 'pending' status for a given account.
+ * @param {String} address - Address of the account to query the transactions from.
+ * @param {Object} configs - Custom configurations for the query (optional).
+ * @param {Object} configs.config - Custom config for the query.
+ * @param {Object} configs.options - Custom options for the query.
+ * @returns - The query state of the API call. Includes the data
+ * (with the array of transactions), loading state, error state, and more.
+ */
+export function useAccountPoolTransactionsQuery(
```
Comment (ManuGowda) [inline src/modules/Accounts/api/useAccountPoolTransactionsQuery.js:15]:
```suggestion
export function useTransactionPoolQuery(
```

maybe name the hook based on endpoint in general?

File: src/modules/Accounts/api/useAccountPoolTransactionsQuery.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ...customConfig,
+    params: {
+      limit: LIMIT,
+      address,
+      executionStatus: 'pending',
+      ...(customConfig?.params || {}),
+    },
+  };
+
+  const keys = useQueryKeys([GET_ACCOUNT_POOL_TRANSACTIONS_QUERY, address, config]);
```
Comment (ManuGowda) [inline src/modules/Accounts/api/useAccountPoolTransactionsQuery.js:32]: same name the constant accordingly to above comment

File: src/modules/Accounts/api/useAccountPoolTransactionsQuery.js (hunk 1 above)
Comment (ClementeSerrano) [reply src/modules/Accounts/api/useAccountPoolTransactionsQuery.js]: done

File: src/modules/Accounts/api/useAccountPoolTransactionsQuery.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const config = {
+    url: `${API_URL}/transactions`,
+    method: 'get',
+    event: 'get.transactions',
+    ...customConfig,
+    params: {
+      limit: LIMIT,
+      address,
+      executionStatus: 'pending',
+      ...(customConfig?.params || {}),
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/Accounts/api/useAccountPoolTransactionsQuery.js]: done

File: src/modules/Accounts/api/useAccountPoolTransactionsQuery.js (hunk 2 above)
Comment (ClementeSerrano) [reply src/modules/Accounts/api/useAccountPoolTransactionsQuery.js]: done

**Review by ManuGowda — DISMISSED** [review]

File: src/modules/Accounts/hooks/useAccountNonce.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    config,
+    options: { cacheTime: 0, ...options },
+  });
+
+  const transactionPoolQueryResult = useTransactionPoolQuery({
+    config: { ...config, params: { address, ...config.params } },
+    options: { cacheTime: 0, ...options },
+  });
+
+  const computeNonce = (_accountAuthQueryResult, _accountPoolTransactionsQueryResult) => {
```
Comment (oskarleonard) [inline src/modules/Accounts/hooks/useAccountNonce.js:25]: - Wouldnt it be more readable if this wasn't scoped inside transactionPoolQueryResult and instead a utility function.
- And have the API something like this `getAccountNounce(transactionsNounce, authNounce)`

File: src/modules/Accounts/hooks/useAccountNonce.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    return poolMaxNonce || onChainNonce;
+  };
+
+  const refetch = async () => {
+    const _accountAuthQueryResult = await accountAuthQueryResult.refetch();
+
+    const _accountPoolTransactionsQueryResult = await transactionPoolQueryResult.refetch();
+
+    return computeNonce(_accountAuthQueryResult, _accountPoolTransactionsQueryResult);
```
Comment (oskarleonard) [inline src/modules/Accounts/hooks/useAccountNonce.js:40]: All return props in the this hook (`useAccountNonce`) follow the default react query implementations except for this refetch, why not just

```
  const refetch = async () => {
    return await Promise.all([...]);
```

File: src/modules/Accounts/hooks/useAccountNonce.js (hunk 4 above)
Comment (ClementeSerrano) [reply src/modules/Accounts/hooks/useAccountNonce.js]: done

**Review by oskarleonard — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**